common/lib/util/defaults.js

```diff
@@ -28,9 +35,10 @@ Defaults.getPort = function(options, tls) {
 
 Defaults.getHosts = function(options) {
 	var hosts = [options.host],
-		fallbackHosts = options.fallbackHosts;
+		fallbackHosts = options.fallbackHosts,
+		httpMaxRetryCount = typeof(options.httpMaxRetryCount) !== 'undefined' ? options.httpMaxRetryCount : Defaults.httpMaxRetryCount;
 
-	if(fallbackHosts) hosts = hosts.concat(fallbackHosts);
+	if(fallbackHosts) hosts = hosts.concat(fallbackHosts.slice(0, httpMaxRetryCount));
```

Comment:
I am not sure this is correct unless the fallback hosts have been randomised first